metadata-ingestion/src/datahub/ingestion/transformer/pattern_cleanup_ownership.py

```diff
@@ -58,11 +61,13 @@ def transform_aspect(
         # clean all the owners based on the parameters received from config
         cleaned_owner_urns: List[str] = []
         for owner_urn in current_owner_urns:
-            user_id: str = owner_urn.split(_USER_URN_PREFIX)[1]
-            for value in self.config.pattern_for_cleanup:
-                user_id = re.sub(value, "", user_id)
-
-            cleaned_owner_urns.append(_USER_URN_PREFIX + user_id)
+            try:
+                user_id: str = owner_urn.split(_USER_URN_PREFIX)[1]
+                for value in self.config.pattern_for_cleanup:
+                    user_id = re.sub(value, "", user_id)
+                cleaned_owner_urns.append(_USER_URN_PREFIX + user_id)
+            except IndexError:
+                log.warning(f"Could not parse {owner_urn} from {entity_urn}")
```

Comment:
can we add some tests that repro the issue?

does it happen when group urns get passed in, or some other situation?

Reply:
mmm yes that's right, it's when it's a group urn. I never actually noticed that pattern 😅